runners/genlayer-py-std/src-emb/genlayer_embeddings/vecdb.py

```
@@ -13,16 +20,54 @@
 
 class Distance(typing.Protocol):
-	def __call__(self, l, r) -> typing.Any: ...
+	"""
+	Protocol for distance functions used by :py:class:`VecDB`.
+
+	Implementations must be a true metric (non-negative, symmetric, zero iff
+	equal, and satisfying the triangle inequality); otherwise the cover-tree
+	pruning in :py:meth:`VecDB.knn` may skip the true nearest neighbor.
+	"""
+
+	def __call__(self, l, r) -> typing.Any:
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)